ui/core/components/suggest_reforges_action.tsx

```diff
@@ -1128,11 +1125,19 @@ export class ReforgeOptimizer {
 							for (const [stat, value] of distributedSocketBonus.entries()) {
 								this.applyReforgeStat(coefficients, stat, value, preCapEPs);
 							}
+						} else if (gemData.gem.requiredProfession > Profession.ProfessionUnknown) {
+							continue;
```

Comment:
Add a comment here explaining that this is a performance optimization to force socket bonus matching for JC gems.